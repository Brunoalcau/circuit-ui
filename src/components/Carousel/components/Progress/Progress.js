import PropTypes from 'prop-types';
import styled, { css, keyframes } from 'react-emotion';

import { SLIDE_DURATION } from '../../constants';

const progressAnimation = keyframes`
  0% {
    width: 0%;
    transform: translateX(0%);
  }

  90% {
    width: 100%;
    transform: translateX(0%);
  }

  100% {
    width: 100%;
    transform: translateX(102%);
  }
`;
const progressStyles = ({ theme, paused, animationDuration }) => css`
  label: carousel__progress;
  width: 100%;
  height: 2px;
  overflow: hidden;
  background-color: ${theme.colors.n200};
  border-radius: ${theme.borderRadius.giga};

  &:after {
    content: '';
    display: block;
    background-color: ${theme.colors.black};
    animation-name: ${progressAnimation};
    animation-duration: ${animationDuration}ms;
    animation-play-state: ${paused ? 'paused' : 'running'};
    animation-fill-mode: forwards;
    height: 100%;
  }
`;
const Progress = styled('div')(progressStyles);

Progress.propTypes = {
  paused: PropTypes.bool,
  animationDuration: PropTypes.number,
  slideDuration: PropTypes.number
};

Progress.defaultProps = {
  /**
   * Indicates if progress animation is currently paused
   */
  paused: false,
  /**
   * Duration of progress animation in milliseconds
   */
  animationDuration: SLIDE_DURATION
};

export default Progress;
