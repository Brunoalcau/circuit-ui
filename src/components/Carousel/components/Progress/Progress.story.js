import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, number } from '@storybook/addon-knobs/react';
import { css } from 'react-emotion';

import { GROUPS } from '../../../../../.storybook/hierarchySeparators';
import withTests from '../../../../util/withTests';

import Progress from './Progress';
import { SLIDE_DURATION } from '../../constants';

storiesOf(`${GROUPS.COMPONENTS}|Carousel/Progress`, module)
  .addDecorator(withTests('Carousel/Progress'))
  .add('Progress', () => (
    <div style={{ width: '60vw' }}>
      <Progress
        css={css`
          &:after {
            animation-iteration-count: infinite;
          }
        `}
        paused={boolean('Paused', false)}
        animationDuration={number('Animation duration', SLIDE_DURATION)}
      />
    </div>
  ));
