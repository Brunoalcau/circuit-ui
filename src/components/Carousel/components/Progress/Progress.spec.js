import React from 'react';

import Progress from './Progress';

describe('Progress', () => {
  describe('styles', () => {
    it('should render with default styles', () => {
      const actual = create(<Progress />);

      expect(actual).toMatchSnapshot();
    });

    it('should render with paused styles', () => {
      const actual = create(<Progress paused />);

      expect(actual).toMatchSnapshot();
    });
  });

  describe('accessibility', () => {
    it('should meet accessibility guidelines', async () => {
      const wrapper = renderToHtml(<Progress />);
      const actual = await axe(wrapper);

      expect(actual).toHaveNoViolations();
    });
  });
});
