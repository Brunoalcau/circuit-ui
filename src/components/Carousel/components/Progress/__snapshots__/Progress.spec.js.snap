// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress styles should render with default styles 1`] = `
.circuit-0 {
  width: 100%;
  height: 2px;
  overflow: hidden;
  background-color: #EEF0F2;
  border-radius: 5px;
}

.circuit-0:after {
  content: '';
  display: block;
  background-color: #0F131A;
  -webkit-animation-name: animation-1sge5un-progressAnimation;
  animation-name: animation-1sge5un-progressAnimation;
  -webkit-animation-duration: 5000ms;
  animation-duration: 5000ms;
  -webkit-animation-play-state: running;
  animation-play-state: running;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  height: 100%;
}

<div
  className="circuit-0 circuit-1"
/>
`;

exports[`Progress styles should render with paused styles 1`] = `
.circuit-0 {
  width: 100%;
  height: 2px;
  overflow: hidden;
  background-color: #EEF0F2;
  border-radius: 5px;
}

.circuit-0:after {
  content: '';
  display: block;
  background-color: #0F131A;
  -webkit-animation-name: animation-1sge5un-progressAnimation;
  animation-name: animation-1sge5un-progressAnimation;
  -webkit-animation-duration: 5000ms;
  animation-duration: 5000ms;
  -webkit-animation-play-state: paused;
  animation-play-state: paused;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  height: 100%;
}

<div
  className="circuit-0 circuit-1"
/>
`;
